front/components/assistant/AgentMessageMarkdown.tsx

```diff
@@ -31,13 +36,20 @@ export const AgentMessageMarkdown = ({
   isStreaming?: boolean;
   additionalMarkdownComponents?: Components;
 }) => {
+  // Preprocess content to handle instruction blocks
+  const processedContent = React.useMemo(
```

Comment:
we need to pre-process instructions blocks as our markdown rendering lib expects `:::` as block starts not XML